src/app/item-popup/PlugTooltip.tsx

```diff
@@ -49,21 +52,84 @@ export default function PlugTooltip({
       )
     : [];
 
+  const stats: { [statHash: string]: number } = {};
+
+  for (const statHash of visibleStats) {
+    const value = plug.stats?.[parseInt(statHash, 10)];
+    if (value) {
+      stats[statHash] = value;
+    }
+  }
+
+  // The PlugTooltip does all the rendering and layout, we just process information here.
+  return (
+    <PlugTooltip
+      def={plug.plugDef}
+      perks={plug.perks}
+      stats={stats}
+      plugObjectives={plug.plugObjectives}
+      enableFailReasons={plug.enableFailReasons}
+      cannotCurrentlyRoll={plug.cannotCurrentlyRoll}
+      wishListTip={wishListTip}
+    />
+  );
+}
+
+/**
+ * This creates a tooltip for a plug with various levels of content.
+ *
+ * It only relies on Bungie API entities, objects or primitives. This is so we can use it to render a
+ * tooltip from either a DimPlug or a DestinyInventoryItemDefinition.
+ *
+ * Use this directly if you want to render a tooltip for a DestinyInventoryItemDefinition, only the def
+ * prop is required.
+ */
+export function PlugTooltip({
+  def,
+  perks,
+  stats,
+  plugObjectives,
+  enableFailReasons,
+  cannotCurrentlyRoll,
+  wishListTip,
+}: {
+  def: DestinyInventoryItemDefinition;
+  perks?: DestinySandboxPerkDefinition[];
+  stats?: { [statHash: string]: number };
+  plugObjectives?: DestinyObjectiveProgress[];
+  enableFailReasons?: string;
+  cannotCurrentlyRoll?: boolean;
+  wishListTip?: string;
+}) {
+  const defs = useD2Definitions();
+  const sourceString =
+    defs && def.collectibleHash && defs.Collectible.get(def.collectibleHash).sourceString;
+
+  let displayedPerks = perks;
+
+  // If perks aren't available from a prop, see if re can get them.
+  if (!displayedPerks) {
+    displayedPerks = _.compact(
+      _.uniqBy(
+        def.perks,
+        (p) => defs?.SandboxPerk.get(p.perkHash).displayProperties.description
+      ).map((perk) => defs?.SandboxPerk.get(perk.perkHash))
+    );
+  }
```

Comment:
One thing to callout is I pushed this all the way down so we can just pass in a def to this component and get a pretty decent tooltip. If we pass in the perks from the `DimPlug` we just use those and skip this step.